From 5afcd7792abfe453be7d4dde6b36032e36f234fb Mon Sep 17 00:00:00 2001
From: zhangmeng <zhangmeng.kevin@spacemit.com>
Date: Tue, 26 Nov 2024 12:26:48 +0800
Subject: [PATCH 1141/1206] wireless: rtl8852be: porting to linux-6.12

1. modify prototype of device remove function
2. replace strlcpy with strscpy

Change-Id: Idd258bb7dfe7183ad658b05a9c7e41b3c659c0b8
---
 .../wireless/realtek/rtl8852be/os_dep/linux/os_intfs.c | 10 +++++-----
 .../realtek/rtl8852be/os_dep/linux/rtw_android.c       |  3 +--
 2 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/drivers/net/wireless/realtek/rtl8852be/os_dep/linux/os_intfs.c b/drivers/net/wireless/realtek/rtl8852be/os_dep/linux/os_intfs.c
index 0f7ff9a8c070..2a2417c708b4 100644
--- a/drivers/net/wireless/realtek/rtl8852be/os_dep/linux/os_intfs.c
+++ b/drivers/net/wireless/realtek/rtl8852be/os_dep/linux/os_intfs.c
@@ -515,13 +515,13 @@ static void rtw_ethtool_get_drvinfo(struct net_device *dev, struct ethtool_drvin
 
 	wdev = dev->ieee80211_ptr;
 	if (wdev) {
-		strlcpy(info->driver, wiphy_dev(wdev->wiphy)->driver->name,
+		strscpy(info->driver, wiphy_dev(wdev->wiphy)->driver->name,
 			sizeof(info->driver));
 	} else {
-		strlcpy(info->driver, "N/A", sizeof(info->driver));
+		strscpy(info->driver, "N/A", sizeof(info->driver));
 	}
 
-	strlcpy(info->version, DRIVERVERSION, sizeof(info->version));
+	strscpy(info->version, DRIVERVERSION, sizeof(info->version));
 
 	padapter = (_adapter *)rtw_netdev_priv(dev);
 
@@ -538,10 +538,10 @@ static void rtw_ethtool_get_drvinfo(struct net_device *dev, struct ethtool_drvin
 	} else
 	#endif
 	{
-		strlcpy(info->fw_version, "N/A", sizeof(info->fw_version));
+		strscpy(info->fw_version, "N/A", sizeof(info->fw_version));
 	}
 
-	strlcpy(info->bus_info, dev_name(wiphy_dev(wdev->wiphy)),
+	strscpy(info->bus_info, dev_name(wiphy_dev(wdev->wiphy)),
 		sizeof(info->bus_info));
 }
 
diff --git a/drivers/net/wireless/realtek/rtl8852be/os_dep/linux/rtw_android.c b/drivers/net/wireless/realtek/rtl8852be/os_dep/linux/rtw_android.c
index 9c12c16285ea..1dea513ad3fd 100644
--- a/drivers/net/wireless/realtek/rtl8852be/os_dep/linux/rtw_android.c
+++ b/drivers/net/wireless/realtek/rtl8852be/os_dep/linux/rtw_android.c
@@ -990,7 +990,7 @@ static void shutdown_card(void)
 }
 #endif /* RTW_SUPPORT_PLATFORM_SHUTDOWN */
 
-static int wifi_remove(struct platform_device *pdev)
+static void wifi_remove(struct platform_device *pdev)
 {
 	struct wifi_platform_data *wifi_ctrl =
 		(struct wifi_platform_data *)(pdev->dev.platform_data);
@@ -1002,7 +1002,6 @@ static int wifi_remove(struct platform_device *pdev)
 	wifi_set_carddetect(0);	/* CardDetect (1->0) */
 
 	up(&wifi_control_sem);
-	return 0;
 }
 
 #ifdef RTW_SUPPORT_PLATFORM_SHUTDOWN
-- 
2.47.0

